scripts/src/primazactl/utils/kubeconfigwrapper.py

```diff
@@ -65,42 +66,61 @@ def get_kube_config_for_cluster(self):
                           "preferences": kcc_yaml["preferences"],
                           "current-context": self.context}
 
-        logger.log_info("look through clusters")
-        for cluster in kcc_yaml["clusters"]:
-            if cluster["name"] == self.context:
-                logger.log_info(f"cluster found: {self.context}")
-                cluster_config["clusters"] = [cluster]
-                break
-
-        logger.log_info("look through users")
-        for user in kcc_yaml["users"]:
-            if user["name"] == self.context:
-                logger.log_info(f"user found: {self.context}")
-                cluster_config["users"] = [user]
-                break
-
-        logger.log_info("look through contexts")
+        context_cluster: str = None
         for context in kcc_yaml["contexts"]:
             if context["name"] == self.context:
                 cluster_config["contexts"] = [context]
                 logger.log_info(f"context found: {self.context}")
+                self.user = context["context"]["user"]
+                context_cluster = context["context"]["cluster"]
                 break
 
-        kcw = KubeConfigWrapper(self.context, None)
+        if self.user != self.context:
+            for context in kcc_yaml["contexts"]:
+                if context["name"] == self.user:
+                    cluster_config["contexts"].append(context)
+                    logger.log_info(f'context found: {context["name"]}')
+                    break
+
+        for cluster in kcc_yaml["clusters"]:
+            if cluster["name"] == self.context or \
+                    (context_cluster and cluster["name"] == context_cluster):
+                logger.log_info(f'cluster found: {cluster["name"]}')
+                cluster_config["clusters"] = [cluster]
+                break
+
+        if "clusters" not in cluster_config:
+            msg = f"Error cluster {self.context} not found in kube config: " \
+                  f"{self.kube_config_file}"
+            logger.log_error(msg)
+            raise RuntimeError(f"[ERROR] {msg}")
+
+        for user in kcc_yaml["users"]:
+            if user["name"] == self.context or \
+                    (self.user and user["name"] == self.user):
+                logger.log_info(f'user found: {user["name"]}')
+                if "users" in cluster_config:
+                    cluster_config["users"].append(user)
+                else:
+                    cluster_config["users"] = [user]
+
+        kcw = KubeConfigWrapper(self.context, self.kube_config_file)
         kcw.kube_config_content = yaml.dump(cluster_config)
-        # logger.log_info(f"Kubeconfig:\n{kcw.kube_config_content}")
+        logger.log_info(f"Kubeconfig:\n{yaml.dump(cluster_config)}")
```

Comment:
Why do log the content of the kubeconfig? What if primaza is run from a workflow, that data will be in the logs forever. Should we log the path instead?

Reply:
Useful for debug, have commented it out again. 
Also we don't print verbose output in our workflows.